packages/flutter/lib/src/material/input.dart

```diff
@@ -85,6 +209,14 @@ class _InputState extends State<Input> {
     ThemeData themeData = Theme.of(context);
     BuildContext focusContext = focusKey.currentContext;
     bool focused = focusContext != null && Focus.at(focusContext, autofocus: config.autofocus);
+    _FormFieldState fieldState = _FormFieldState.at(context, config.formField, this);
+    InputValue value = fieldState?.value ?? config.value;
+    ValueChanged<InputValue> onChanged = fieldState?.onChanged ?? config.onChanged;
+    ValueChanged<InputValue> onSubmitted = fieldState?.onSubmitted ?? config.onSubmitted;
```

Comment:
Good question - I was thinking that any Input inside of a Form would always be in Form mode, and we'd completely ignore the constants. But maybe it would be useful to have loose Inputs inside a Form. I've changed it.